server/src/main/java/org/candlepin/dto/api/v1/CertificateSerialTranslator.java

```diff
@@ -56,14 +61,26 @@ public CertificateSerialDTO populate(CertificateSerial source, CertificateSerial
     @Override
     public CertificateSerialDTO populate(ModelTranslator translator, CertificateSerial source,
         CertificateSerialDTO dest) {
+        if (source == null) {
+            throw new IllegalArgumentException("source is null");
+        }
 
-        dest = super.populate(translator, source, dest);
+        if (dest == null) {
+            throw new IllegalArgumentException("dest is null");
+        }
 
-        dest.setId(source.getId() == null ? null : source.getId().toString());
-        dest.setSerial(source.getSerial() == null ? null : source.getSerial().toString());
-        dest.setExpiration(source.getExpiration());
-        dest.setCollected(source.isCollected());
-        dest.setRevoked(source.isRevoked());
+        Date created = source.getCreated();
+        dest.setCreated(created != null ?  Instant.ofEpochMilli(created.getTime()).atZone(ZoneId.systemDefault()).toLocalDate(): null);
```

Comment:
I wonder if it'd be worth the effort (and if it's even possible) to update the entities to use LocalDate as well. While this translation works, I don't know how keen I am to be doing that everywhere.

Reply:
I tried looking into changing CertificateSerial.java's expiration field from Date to OffsetDateTime.
We would need to change our CertificateSerialInfo Adapter inferface to support OffsetDateTime too, which would also force us to change the manifest CertificateSerialDTO to use it too, and this whole thing suddenly cascades wildly. Basically, if we change our model Dates to OffsetDateTimes, the Dates on the *Info would all also need to change. Not only that, but Date is basically used all across the codebase, so if we start changing those on the models, we'd have to change them everywhere (including model unit tests).

Compared to all that, doing these conversions in translators only (and translator tests), is a good price to pay. So I'd vote to not do that. If we _were_ going to do it though, it would probably have to be a massive effort, which I'd prefer be done separately alltogether from this epic (probably another epic).

Reply:
Yikes, alright then.

We probably should switch to the java.time package eventually, but if it's going to be such a massive undertaking, then I agree that this PR is not the place for it.